x-pack/plugins/fleet/public/applications/integrations/hooks/use_package_install.tsx

```diff
@@ -26,13 +29,16 @@ interface PackagesInstall {
 interface PackageInstallItem {
   status: InstallStatus;
   version: string | null;
+  error?: FleetErrorResponse;
 }
 
 type InstallPackageProps = Pick<PackageInfo, 'name' | 'version' | 'title'> & {
   isReinstall?: boolean;
   fromUpdate?: boolean;
+  force?: boolean;
 };
-type SetPackageInstallStatusProps = Pick<PackageInfo, 'name'> & PackageInstallItem;
+type SetPackageInstallStatusProps = Pick<PackageInfo, 'name'> &
+  PackageInstallItem & { error?: FleetErrorResponse };
 
 function usePackageInstall({
```

Comment:
Yeah I'd be interested to see if we can have a "global" modal instead of rendering it in a few different places. The hook can then toggle some context value to hide/show the global modal. Similar to how some of our flyouts work. The overlay service @nchaulet mentioned is also probably a good idea, but I haven't had experience using it yet.

Reply:
an example on how we use the overlay service in Fleet already, it also make testing the hook quite easy, you just have to mock that service https://github.com/hop-dev/kibana/blob/c73ea10d74cc5210dcdbae5ec4177175f49ecd94/x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/edit_package_policy_page/hooks/index.tsx#L25

Reply:
I agree that looks better, I'll have a go 👍

Reply:
Seems like this went mostly well. Any gotchas we should be aware of here?